File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    for name, kind in clothes:
        if kind in dict:
            dict[kind] += [name]
        else:
            dict[kind] = [name]
    for _, value in dict.items():
        answer *= (len(value) + 1)
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    const clothMap = {};
    
    for (const c of clothes) {
        const [name, type] = c;
        clothMap[type] = (clothMap[type] || 0) + 1
    }
    
    for (const type in clothMap) {
        answer *= (clothMap[type] + 1)
    }
    
    return answer - 1;
}